src/python/pants/engine/legacy/address_mapper.py

```diff
@@ -56,8 +56,14 @@ def is_declaring_file(address, file_path):
     # information for debugging most of the time.
     #
     # We could call into the engine to ask for the file that declared the address.
-    return (os.path.dirname(file_path) == address.spec_path and
-            BuildFile._is_buildfile_name(os.path.basename(file_path)))
+    if not BuildFile._is_buildfile_name(os.path.basename(file_path)):
```

Comment:
part of it still applies to "except" block, I will move it there and rephrase it.